### ISSUE TITLE
the test result is 0

### ISSUE THREAD

**Issue by AnLingla** [yzd-v/FGD#9]
Dear author, I configured my training model on MMDetection, and the bbox_mAP_50 of the validation set was preferably 0.883, but the test result was 0.My detailed steps are as follows:
**Add and Replace the codes**
1.Add folders a_my_configs/ to the configs/ in mmdetectin's codes.
2.Add my_faster_rcnn_r50_fpn_2x_coco.py and my_faster_rcnn_r101_fpn_2x_coco.py to the configs/a_my_configs/.
3.In my_faster_rcnn_r50_fpn_2x_coco.py,   _base_ as same as configs/faster_rcnn/faster_rcnn_r50_fpn_2x_coco.py, add 
data=dict(
    samples_per_gpu=1,  # batch size
    workers_per_gpu=1,  # num_workers
    train=dict(path1),val=dict(path2),test=dict(path3))
4.Modify the  
student_cfg = 'configs/a_my_configs/my_faster_rcnn_r50_fpn_2x_coco.py'
teacher_cfg = 'configs/a_my_configs/my_faster_rcnn_r101_fpn_2x_coco.py' 
in configs/distillers/fgd/fgd_faster_rcnn_r101_fpn_2x_distill_faster_rcnn_r50_fpn_2x_coco.py
**Train**
python tools/train.py configs/distillers/fgd/fgd_faster_rcnn_r101_fpn_2x_distill_faster_rcnn_r50_fpn_2x_coco.py
The training process went smoothly and the best bbox_mAP_50=0.883
**Test**
python tools/test.py configs/a_my_configs/my_faster_rcnn_r50_fpn_2x_coco.py work_dirs/fgd_faster_rcnn_r101_fpn_2x_distill_faster_rcnn_r50_fpn_2x_coco/latest.pth --eval bbox

But the test results(bbox_mAP,bbox_mAP_50,bbox_mAP_l……) were 0, I guess the problem appears on the configuration file, but do not know how to solve, looking forward to your reply, I can provide my complete code

**Comment by yzd-v** [comment]
You have to transfer the FGD model into MMDet model because they have different keys. You can use pth_transfer.py to transfer it. The details can be seen in README Transfer.

**Comment by AnLingla** [comment]
I am so happy to receive your reply. Depending on the method you provided, the testing issue has been solved. 
But I still have another question, can I combine two algorithms of different architectures in the mmdetection you provided?
For example, use swin transformer as the teacher model and rnn50 as the student model

**Comment by AnLingla** [comment]
I am so happy to receive your reply. Depending on the method you provided, the testing issue has been solved. But I still have another question, can I combine two algorithms of different architectures in the mmdetection you provided?For example, use swin transformer as the teacher model and rnn50 as the student model

-----原始邮件-----
发件人:Mesopotamia ***@***.***>
发送时间:2022-03-16 14:34:04 (星期三)
收件人: yzd-v/FGD ***@***.***>
抄送: AnLingla ***@***.***>, Author ***@***.***>
主题: Re: [yzd-v/FGD] the test result is 0 (Issue #9)






Closed #9.

—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by yzd-v** [comment]
If the teacher and student have the same H,W of the feature, FGD can be applied successfully. However, transformer and CNN are very different. You can try it by using swin transformer to distill Res-50.

**Comment by AnLingla** [comment]
OK，thanks for your advice
